PWGHF/HFC/Tasks/taskCorrelationLcHadrons.cxx

```diff
@@ -0,0 +1,271 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file taskCorrelationLcHadrons.cxx
+/// \brief Lc-Hadrons azimuthal correlations analysis task - data-like, Mc-Reco and Mc-Gen analyses
+/// \author Marianna Mazzilli <marianna.mazzilli@cern.ch>
+/// \author Zhen Zhang <zhenz@cern.ch>
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "Framework/runDataProcessing.h"
+
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+#include "PWGHF/Utils/utilsAnalysis.h"
+#include "PWGHF/HFC/DataModel/CorrelationTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+///
+/// Returns deltaPhi value in range [-pi/2., 3.*pi/2], typically used for correlation studies
+///
+double getDeltaPhi(double phiLc, double phiHadron)
+{
+  return RecoDecay::constrainAngle(phiHadron - phiLc, -o2::constants::math::PIHalf);
+}
+
+///
+/// Returns phi of candidate/particle evaluated from x and y components of segment connecting primary and secondary vertices
+///
+double evaluatePhiByVertex(double xVertex1, double xVertex2, double yVertex1, double yVertex2)
+{
+  return RecoDecay::phi(xVertex2 - xVertex1, yVertex2 - yVertex1);
+}
+
+// string definitions, used for histogram axis labels
+const TString stringPtLc = "#it{p}_{T}^{#Lambda_c} (GeV/#it{c});";
+const TString stringPtHadron = "#it{p}_{T}^{Hadron} (GeV/#it{c});";
+const TString stringPoolBin = "poolBin;";
+const TString stringDeltaEta = "#it{#eta}^{Hadron}-#it{#eta}^{#Lambda_c};";
+const TString stringDeltaPhi = "#it{#varphi}^{Hadron}-#it{#varphi}^{#Lambda_c} (rad);";
+const TString stringLcHadron = "#Lambda_c,Hadron candidates ";
+const TString stringSignal = "signal region;";
+const TString stringSideband = "sidebands;";
+const TString stringMcParticles = "Mc gen - #Lambda_c,Hadron particles;";
+const TString stringMcReco = "Mc reco - #Lambda_c,Hadron candidates ";
+
```

Comment:
Add a comment for the objects defined below.

Reply:
ok, thanks a lot!